classes/fullstack/project1/client/src/store/slices/auth.slice.ts

```diff
@@ -0,0 +1,21 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+import { Admin } from "../../types"
+
+type InitialState = Pick<Admin, 'username'>
```

Comment:
lets include all the attributes of `AdminRegister` here and give the type a better descriptive name `AuthState` otherwise wehn trying to infer this might be very confusing.

Reply:
done